src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -206,135 +221,184 @@ public static Date convertUtcToLocalDate(Date utcDate, double timeZone) {
      * hour just after midnight is converted to option 24 (i.e., 2359 as shown
      * to the user) 23.59 is also converted to 24. (i.e., 23.59-00.59 ---> 24)
      */
+    @Deprecated
     public static int convertToOptionValueInTimeDropDown(Date date) {
+        return convertToOptionValueInTimeDropDown(convertDateToLocalDateTime(date));
+    }
+
+    /**
+     * Formats a date in the corresponding option value in 'Time' dropdowns The
+     * hour just after midnight is converted to option 24 (i.e., 2359 as shown
+     * to the user) 23.59 is also converted to 24. (i.e., 23.59-00.59 ---> 24)
+     */
+    public static int convertToOptionValueInTimeDropDown(LocalDateTime localDateTime) {
         //TODO: see if we can eliminate this method (i.e., merge with convertToDisplayValueInTimeDropDown)
-        Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
-        c.setTime(date);
-        int hour = c.get(Calendar.HOUR_OF_DAY);
-        int minutes = c.get(Calendar.MINUTE);
-        hour = hour == 0 ? 24 : hour;
-        hour = hour == 23 && minutes == 59 ? 24 : hour;
-        return hour;
+        int hour = localDateTime.getHour();
+        if (hour == 0) {
+            return 24;
```

Comment:
In this case when the else part has only 1 line, I don't think it makes much difference. But if you think it's easier to read then I'll change.

Reply:
There are multiple `else`s here. If I was trying to change this code in the future, it would be a pain to have to mentally invert all the previous if statements in order to figure out what it's doing. Early returns breaks this down into multiple parts which are much easier to parse. It'll functionally be the same but one style is much easier to maintain than the other :smile: 

http://blog.timoxley.com/post/47041269194/avoid-else-return-early
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement